saidump/saidump.cpp

```diff
@@ -399,7 +417,170 @@ void dumpGraph(const TableDump& td)
+        // Close the input file
+        input_file.close();
+        return SAI_STATUS_FAILURE;
+    }
+
+    input_file.seekg(0, std::ios::beg);     // Move to the begin of the file
+
+    // Read the content of the input file into a string
+    std::string content((std::istreambuf_iterator<char>(input_file)),
+                   std::istreambuf_iterator<char>());
+    input_file.close();
+
+    content = regex_replace(content, std::regex("\\},\\{\\r"), ",");
+
+    //erase the 1st and last char.
+    if (content.length() >= 2 && content[0] == '[' && content[content.length()-1] == ']')
+    {
+        content.erase(0, 1);
+        content.erase(content.length() - 1);
+    }
+
+    std::ofstream outputFile(file_name);
+
+    if (!outputFile.is_open())
+    {
+        std::cerr << "Failed to open the output file " << file_name << std::endl;
+        SWSS_LOG_ERROR("Failed to open the output file %s.", file_name.c_str());
+        return SAI_STATUS_FAILURE;
+    }
+
+    outputFile << content;
+    outputFile.close();
+
+    return SAI_STATUS_SUCCESS;
+}
+
+int dump_from_redis_rdb_file(std::string file_name)
+{
+    std::ifstream input_file(file_name);
+
+    if (!input_file.is_open())
+    {
+        std::cerr << "Failed to open the input file " << file_name << std::endl;
+        SWSS_LOG_ERROR("The file %s does not exist for dumping from redis dump.rdb file.", file_name.c_str());
+        return SAI_STATUS_FAILURE;
+    }
+
+    input_file.seekg(0, std::ios::end);     // Move to the end of the file
+    int64_t file_size = input_file.tellg(); // Get the current position
+
+    if (file_size >= RDB_FILE_MAX_SIZE)
+    {
+        std::cerr << "Get " << file_name.c_str() << "'s size failure or its size " << file_size << "  >= "<< RDB_FILE_MAX_SIZE / 1024 / 1024<<" MB." << std::endl;
+        SWSS_LOG_ERROR("Get %s's size failure or its size %ld  >= %ld MB.", file_name.c_str(), file_size, RDB_FILE_MAX_SIZE / 1024 / 1024);
+
+        // Close the input file
+        input_file.close();
+        return SAI_STATUS_FAILURE;
+    }
+
+    input_file.seekg(0, std::ios::beg);     // Move to the begin of the file
+
+    try
+    {
+        // Parse the JSON data from the file (validation)
+        nlohmann::json jsonData;
+        input_file >> jsonData;
+        input_file.close();
+
+        SWSS_LOG_DEBUG("JSON file is valid.");
+
+        for (json::iterator it = jsonData.begin(); it != jsonData.end(); ++it)
+        {
+            json jj_key = it.key();
+
+            std::string keystr = jj_key;
+            std::string item_name = keystr;
+            size_t pos = keystr.find_first_of(":");
+
+            if (pos != std::string::npos)
+            {
+                if(ASIC_STATE_TABLE != keystr.substr(0, pos))  // filter out non ASIC_STATE
+                {
+                    continue;
+                }
+                item_name = keystr.substr(pos + 1);
```

Comment:
empty line after } and before if, please keep this in entire file

Reply:
Fix it.